openedx/stanford/djangoapps/register_cme/forms.py

```diff
@@ -139,22 +146,23 @@ def clean_affiliation(self):
         affiliation = self.cleaned_data['affiliation']
         if affiliation != 'Stanford University':
             return affiliation
-        fields = [
-            {
-                'sunet_id': self.fields['sunet_id'].error_messages['invalid'],
-            },
-            {
-                'stanford_department': self.fields['stanford_department'].error_messages['invalid'],
-            },
-        ]
-        for field in fields:
-            for key, value in field.iteritems():
-                if len(data.get(key)) < 2:
-                    self.add_error(key, value)
-                    raise forms.ValidationError(
-                        value,
-                        code='invalid',
-                    )
+        errors = []
+        sunet_id = self.data['sunet_id']
+        stanford_department = self.data['stanford_department']
+        if not(sunet_id):
+            errors.append({'sunet_id', 'required'})
+        elif (2 < sunet_id < 9):
```

Comment:
This looks like it should check len(sunet_id).  But then the condition would be throwing an error if the length is between 2 and 9.  Which is opposite of what we want.